### ISSUE TITLE
Closing app in "recent apps" screen does not remove cookies (and maybe more)

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/focus-android#1875]
Android 8.1 Beta, Pixel XL.

STR:
* Got to twitter.com
* Login
* Go to "recent apps" screen
* Close Focus by swiping it away
* Open Focus again
* Go to twitter.com

Expected: You are not logged in anymore.

Actual: You are still logged in.

We handle this situation in SessionNotificationService.onTaskRemoved():
https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/session/SessionNotificationService.java#L81

**Comment by mcomella** [comment]
It's hard to track this down entirely but it looks like (using call hierarchies):

1. **Delete everything** when session closed by:
    1. Back pressed
    1. Click on trash can
```
SystemWebView.cleanup - deletes everything
  BrowserFragment.erase
    BF.onBackPressed
    BF.onClick (trash can)
```

2. **Delete WebView files** when:
    1. (the case above: back pressed or trash can click)
    1. All sessions cleared
    1. When MainActivity finishes
    1. WebView is destroyed
```
SystemWebView.deleteContentFromKnownLocations - delete WebView files
  SystemWebView.cleanup (from the above)
  SystemWebView.destroy - when WebView is removed from memory
  WebViewProvider.performCleanup
    MainActivity.onPause (when finishing)
    CleanupSessionObserver.onValueChanged (no more sessions)
```

3. **Delete LocalStorage** when:
    1. We move from 0 sessions to 1 session
```
WebViewProvider.performNewBrowserCleanup() - Deletes LocalStorage
  sessionObserver.onValueChanged (set in MainActivity.onCreate)
```

To further follow when 2) occurs, we should identify when all the sessions are removed.

---

The code that Sebastian points out removes the sessions when the NotificationService task is killed. We should verify:
- Cookies/logins are removed when the WebView files are deleted
- NotificationService.onTaskRemoved is getting called before the process closes
- Should cookies get removed when webkit files

**Comment by mcomella** [comment]
> Cookies/logins are removed when the WebView files are deleted

This doesn't happen. To test this, I changed the trash can just to remove these files, clicked refresh, and I'm still logged in.

We can fix this by deleting the cookies at the start of a new browser session, like we did for #1306.

There may be other hidden issues we're not fixing here – we should try to comprehensively fix this issue in #1569.

**Comment by mcomella** [comment]
fwiw, I was also able to reproduce this with Spotify login.
